banzai/photometry.py

```diff
@@ -241,3 +243,40 @@ def do_stage(self, image):
         except Exception:
             logger.error(logs.format_exception(), image=image)
         return image
+
+
+class PhotometricCalibrator(Stage):
+    color_to_fit = {'gp': 'g-r', 'rp': 'r-i', 'ip': 'r-i', 'zs': 'i-z'}
+
+    def __init__(self, runtime_context):
+        super(PhotometricCalibrator, self).__init__(runtime_context)
+
+    def do_stage(self, image):
+        if image.filter not in ['gp', 'rp', 'ip', 'zs']:
+            return image
+
+        try:
+            # Get the sources in the frame
+            reference_catalog = get_reference_sources(image.meta, self.runtime_context.REFERENCE_CATALOG_URL)
+        except HTTPError as e:
+            logger.error(f'Error retrieving photometric reference catalog: {e}', image=image)
+            return image
+
+        # Match the catalog to the detected sources
+        matched_catalog = match_catalogs(image.catalog, reference_catalog)
+
+        # catalog_mag = instrumental_mag + zeropoint + color_coefficient * color
+        # Fit the zeropoint and color_coefficient rejecting outliers
+        # Note the zero index here in the filter name is because we only store teh first letter of the filter name
+        zeropoint, zeropoint_error, color_coefficient, color_coefficient_error = \
+            fit_photometry(matched_catalog, image.filter[0], self.color_to_fit[image.filter], image.exptime)
+        # Save the zeropoint, color coefficient and errors to header
+        image.meta['L1ZP'] = zeropoint, "Instrumental zeropoint [mag]"
```

Comment:
Possibly add "color-corrected" to key word description is character limit allows. 

The zeropoint as presented is for stars with a reference color==0, and only for star-like spectral energy distributions.  The danger is to apply a blanket zeropoint to all sources and think we are done - that is not the case. 

At least the BANZI web page (is thereone? read the docs?) should have a comment to this end.

Reply:
My hope is that since there is explicitly a color term in the header that it should not be confusing. We should indeed add more high level description to the docs though.